### ISSUE TITLE
AutoDetectingStackNameProvider Fails For Auto-Scaling Groups/ELB

### ISSUE THREAD

**Issue by darylrobbins** [spring-cloud/spring-cloud-aws#50]
Trying to determine the stack name for an instance fails when the instance is connected to the stack via an auto-scaling group as opposed to directly, resulting in an exception that fails the application loading. This exception should likely be swallowed up, allowing the name detector to move on to its fallback logic.

```
Caused by: com.amazonaws.AmazonServiceException: Stack for i-9ae92175 does not exist (Service: AmazonCloudFormation; Status Code: 400; Error Code: ValidationError; Request ID: 4ccbd832-ad6f-11e4-8f95-617fd8834716)
	at com.amazonaws.http.AmazonHttpClient.handleErrorResponse(AmazonHttpClient.java:1071)
	at com.amazonaws.http.AmazonHttpClient.executeOneRequest(AmazonHttpClient.java:719)
	at com.amazonaws.http.AmazonHttpClient.executeHelper(AmazonHttpClient.java:454)
	at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:294)
	at com.amazonaws.services.cloudformation.AmazonCloudFormationClient.invoke(AmazonCloudFormationClient.java:1078)
	at com.amazonaws.services.cloudformation.AmazonCloudFormationClient.describeStackResources(AmazonCloudFormationClient.java:875)
	at org.springframework.cloud.aws.core.env.stack.config.AutoDetectingStackNameProvider.autoDetectStackName(AutoDetectingStackNameProvider.java:63)
	at org.springframework.cloud.aws.core.env.stack.config.AutoDetectingStackNameProvider.afterPropertiesSet(AutoDetectingStackNameProvider.java:53)
	at org.springframework.cloud.aws.core.env.stack.config.AutoDetectingStackNameProvider.<init>(AutoDetectingStackNameProvider.java:44)
	at org.springframework.cloud.aws.core.env.stack.config.AutoDetectingStackNameProvider.<init>(AutoDetectingStackNameProvider.java:48)
	at org.springframework.cloud.aws.core.env.stack.config.StackResourceRegistryFactoryBean.<init>(StackResourceRegistryFactoryBean.java:57)
	at org.springframework.cloud.aws.autoconfigure.context.ContextStackAutoConfiguration$StackAutoDetectConfiguration.stackResourceRegistryFactoryBean(ContextStackAutoConfiguration.java:63)
	at org.springframework.cloud.aws.autoconfigure.context.ContextStackAutoConfiguration$StackAutoDetectConfiguration$$EnhancerBySpringCGLIB$$98368e1.CGLIB$stackResourceRegistryFactoryBean$0(<generated>)
	at org.springframework.cloud.aws.autoconfigure.context.ContextStackAutoConfiguration$StackAutoDetectConfiguration$$EnhancerBySpringCGLIB$$98368e1$$FastClassBySpringCGLIB$$1351ccc2.invoke(<generated>)
	at org.springframework.cglib.proxy.MethodProxy.invokeSuper(MethodProxy.java:228)
	at org.springframework.context.annotation.ConfigurationClassEnhancer$BeanMethodInterceptor.intercept(ConfigurationClassEnhancer.java:309)
	at org.springframework.cloud.aws.autoconfigure.context.ContextStackAutoConfiguration$StackAutoDetectConfiguration$$EnhancerBySpringCGLIB$$98368e1.stackResourceRegistryFactoryBean(<generated>)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:162)
	... 89 more
```

**Comment by aemruli** [comment]
@darylrobbins can you elaborate more "This exception should likely be swallowed up, allowing the name detector to move on to its fallback logic." what you mean with that ?

**Comment by sodik82** [comment]
is there at least any workaround for this problem?

**Comment by sodik82** [comment]
for me (I am not using CloudFormation) worked spring-boot property:

cloud.aws.stack.auto = false

**Comment by aemruli** [comment]
@sodik82 You can also specify the stack name if your are using CloudFormation. We will have a investigation into this bug.

**Comment by sodik82** [comment]
I have tried that as well, but in such case I got exception saying that stack name was not found (that was quite acceptable since I don't use CloudFormation at all). But maybe we are talking about two different problems here:
1. my case - spring boot autoconfiguration, not using CloudFormation, no ELB
2. reporter's case - (if I understand it correctly): EC2 instance behind ELB, maybe using CloudFormation

however I got here since I get (at least very similar) exception with "default" spring boot autoconfiguration which was working as long I was on my local machine but not after deployment to AWS. can I help somehow with more details?

**Comment by darylrobbins** [comment]
@sodik82 is correct. My configuration was EC2 instance in an autoscaling group behind an ELB, managed by Elastic Beanstalk, which uses CloudFormation under the hood.

**Comment by sodik82** [comment]
@aemruli shall I create an extra ticket?

**Comment by unniksk** [comment]
Hi,

I am getting the same error when deploying the Spring boot jar using Elastic Beanstalk but the same jar is working on local.
Is there any way to resolver this issue.

**Comment by aemruli** [comment]
There was an error with the auto configuration for the region therefore the stacks were not found. The merged #115 should solve that issue. 

Please reopen in case of any further error.
